tests/org.eclipse.cdt.lsp.clangd.tests/src/org/eclipse/cdt/lsp/internal/clangd/tests/ClangdConfigFileCheckerTest.java

```diff
@@ -132,15 +133,19 @@ void testInvalidYamlSyntaxMissingBrace() throws IOException, CoreException {
 		// GIVEN an existing .clangd configuration file with invalid yaml syntax (missing closing brace):
 		// WHEN a new .clang file gets added to the project (should trigger ClangdConfigFileMonitor.checkJob).
 		var configFile = createConfigFile(INVALID_YAML_SYNTAX_MISSING_BRACE);
-		// AND the UI thread sleeps for 1s:
-		try {
-			Thread.sleep(1000);
-		} catch (InterruptedException e) {
-			e.printStackTrace();
+		int cnt = 0;
+		var marker = new IMarker[] {};
+		while (marker.length == 0 && cnt < 20) {
+			marker = configFile.findMarkers(ClangdConfigFileChecker.CLANGD_MARKER, false, IResource.DEPTH_ZERO);
+			cnt++;
+			try {
+				Thread.sleep(50);
+			} catch (InterruptedException e) {
+				e.printStackTrace();
+			}
```

Comment:
In test you can just let the InterruptedException be thrown by the test method. It makes the test shorter, and if a thread is interrupted here that would be unexpected, so having the stacktrace saved in the test results (instead of just stderr) is good.

```suggestion
			Thread.sleep(50);
```